### PR TITLE
【Feature】fix the resolution problem for clip-vision transformer part and swim …

### PR DESCRIPTION
**Pull request by psky1111**
fix the resolution problem for clip vision transformer part and swim transformer
For swin transformed the main change is patchmerged, patchembed and atten mask.
For clip vision transformer the main change in patchembed.

### CODE REVIEW

File: ppcls/arch/backbone/model_zoo/foundation_vit.py [hunk 1]
```diff
@@ -499,10 +499,28 @@ def __init__(self, img_size=224, patch_size=16, in_chans=3, embed_dim=768):
             stride=patch_size,
             bias_attr=False)
 
+    def maybe_pad(self, pixel_values, height, width):
```
Comment (cuicheng01) [inline ppcls/arch/backbone/model_zoo/foundation_vit.py]: 函数名换个名字吧

File: ppcls/arch/backbone/model_zoo/foundation_vit.py [hunk 2]
```diff
@@ -499,10 +499,28 @@ def __init__(self, img_size=224, patch_size=16, in_chans=3, embed_dim=768):
...
+        if width % self.patch_size[1] != 0:
+            pad_values = (0, 0, 0, 0, 0, 0, 0,
+                          self.patch_size[1] - width % self.patch_size[1])
+            pixel_values = nn.functional.pad(pixel_values, pad_values)
+        if height % self.patch_size[0] != 0:
+            pad_values = (
+                0,
```
Comment (cuicheng01) [inline ppcls/arch/backbone/model_zoo/foundation_vit.py]: 文件需要过一下pre-commit

File: ppcls/arch/backbone/model_zoo/swin_transformer_v2.py [hunk 3]
```diff
@@ -370,17 +375,62 @@ def __init__(self,
...
             attn_mask = masked_fill(attn_mask, attn_mask == 0, float(0.0))
         else:
-            attn_mask = None
+        """
+        H, W = self.input_resolution
+        attn_mask = paddle.zeros([1, H, W, 1])
 
         self.register_buffer("attn_mask", attn_mask)
 
+    def maybe_pad(self, hidden_states, height, width):
```
Comment (cuicheng01) [inline ppcls/arch/backbone/model_zoo/swin_transformer_v2.py]: 同上

File: ppcls/arch/backbone/model_zoo/swin_transformer_v2.py [hunk 4]
```diff
@@ -605,11 +681,28 @@ def __init__(self,
         else:
             self.norm = None
 
+    def maybe_pad(self, pixel_values, height, width):
```
Comment (cuicheng01) [inline ppcls/arch/backbone/model_zoo/swin_transformer_v2.py]: 不同类中的maybe_pad函数能否统一一下

**Comment by psky1111** [comment]
image
image
基于256*256 pretrain model, 使用242*242解析度验证模型, 以上为验证logs以及论文中swin-base acc. 242的acc优于论文中的acc

**Comment by psky1111** [comment]
swim v1验证acc logs:
eval 日志
image
infer日志
image

**Comment by psky1111** [comment]
基于原224*224 logs
image

**Review by cuicheng01 — APPROVED** [review]
